content/documentation/master/batching.md

```diff
@@ -272,14 +318,77 @@ The following is how you can still have asynchronous code, by placing it into th
 functions they all can run at once.  In the code above `CompletableFuture.supplyAsync(() -> getTheseCharacters(keys));` will run the ``getTheseCharacters``
-method in another thread.
\ No newline at end of file
+method in another thread.
+
+# Passing context to your data loader
+
+The data loader library supports two types of context being passed to the batch loader. The first is
+an overall context object per dataloader and the second is a map of per loaded key context objects.
+
+This allows you to pass in the extra details you may need to make downstream calls.  The dataloader key is used
+in the caching of results but the context objects can be made available to help with the call.
+
+So in the example below we have an overall security context object that gives out a call token and we also pass the graphql source
+object to each ``dataLoader.load()`` call.
+
+{{< highlight java "linenos=table" >}}
+
+        BatchLoaderWithContext<String, Object> batchLoaderWithCtx = new BatchLoaderWithContext<String, Object>() {
+
+            @Override
+            public CompletionStage<List<Object>> load(List<String> keys, BatchLoaderEnvironment loaderContext) {
+                //
+                // we can have an overall context object
+                SecurityContext securityCtx = loaderContext.getContext();
+                //
+                // and we can have a per key set of context objects
+                Map<Object, Object> keysToSourceObjects = loaderContext.getKeyContexts();
+
+                return CompletableFuture.supplyAsync(() -> getTheseCharacters(securityCtx.getToken(), keys, keysToSourceObjects));
+            }
+        };
+
+        // ....
+
+        SecurityContext securityCtx = SecurityContext.newSecurityContext();
+
+        BatchLoaderContextProvider contextProvider = new BatchLoaderContextProvider() {
+            @Override
+            public Object getContext() {
+                return securityCtx;
+            }
+        };
+        //
+        // this creates an overall context for the dataloader
+        //
+        DataLoaderOptions loaderOptions = DataLoaderOptions.newOptions().setBatchLoaderContextProvider(contextProvider);
+        DataLoader<String, Object> characterDataLoader = DataLoader.newDataLoader(batchLoaderWithCtx, loaderOptions);
+
+        // .... later in your data fetcher
+
+        DataFetcher dataFetcherThatCallsTheDataLoader = new DataFetcher() {
+            @Override
+            public Object get(DataFetchingEnvironment environment) {
+                String argId = environment.getArgument("id");
+                Object source = environment.getSource();
+                //
+                // you can pass per load call contexts
+                //
+                return characterDataLoader.load(argId, source);
+            }
+        };
+        
+{{< / highlight >}}
+
```

Comment:
Should these changes be copied over to other documentation directories (v10 and v9)?

Reply:
no - master only